File: test-metering.py
import boto3
import time

dynamodb = boto3.resource('dynamodb', region_name='us-east-1')
table = dynamodb.Table('mp-anycompany-MeteringRecords')

# Three records with bad dimensions to trigger failures
for i, dim in enumerate(['BadDimension1', 'BadDimension2', 'BadDimension3']):
    table.put_item(Item={
        'customerAWSAccountId': '678222115656',
        'create_timestamp': int(time.time() * 1000) + i,
        'productCode': '21bee1a048vloys14ivpxn0np',
        'dimension': dim,
        'quantity': 10 + i,
        'metering_pending': 'true'
    })
    print(f'Written: {dim}')

print('All 3 test records written')

File: test-metering.mjs
import { DynamoDBClient } from '@aws-sdk/client-dynamodb';
import { DynamoDBDocumentClient, PutCommand } from '@aws-sdk/lib-dynamodb';
const dynamodb = DynamoDBDocumentClient.from(new DynamoDBClient({ region: 'us-east-1' }));

await dynamodb.send(new PutCommand({
    TableName: 'mp-anycompany-MeteringRecords',
    Item: {
        customerAWSAccountId: '123456789012',
        create_timestamp: Date.now(),
        productCode: 'prod-abc123',
        dimension: 'ApiCalls',
        quantity: 150,
        metering_pending: 'true'
    }
}));

console.log('Usage record written successfully');
